### DISCUSSION TITLE
The best free packet for creating reports in Avalonia UI

### DISCUSSION THREAD

**Discussion by vo1s** [AvaloniaUI/Avalonia#14244] [Q&A] [answered]
I am currently choosing the packet for creating reports in Avalonia. I found Stimulsoft.Reports but as far as i know you have to pay for this. Is there any other packages for this purpose?

**Comment by thevortexcloud** [comment]
If you don't need anything particularly fancy you can easily make your own using something like QuestPDF. But I also guess it depends on exactly what you are looking for.

**Reply by vo1s** [reply]
I need to create pdf files from templates

**Reply by thevortexcloud** [reply, marked as answer]
QuestPDF can do that. The templates are just defined in C#. Once you get it set up you just pass your model into your generator function and it will output a PDF. They have an example here for how to generate an invoice:

https://www.questpdf.com/getting-started.html

You can export the PDF to an image and then display it in Avalonia as a bitmap. I have done it myself in the past, it works pretty well. I had it automatically updating with my UI so it always matched what ever data I was entering.

I should note that QuestPDF is only available for free under the MIT license for individuals and small businesses. Large businesses need to pay.

**Reply by vo1s** [reply]
Thank you.
